matchengine/me_market.c

```diff
@@ -776,24 +776,32 @@ static int execute_market_bid_order(bool real, market_t *m, order_t *taker)
         order_t *maker = node->value;
         mpd_copy(price, maker->price, &mpd_ctx);
 
-        mpd_div(amount, taker->left, price, &mpd_ctx);
-        mpd_rescale(amount, amount, -m->stock_prec, &mpd_ctx);
-        while (true) {
-            mpd_mul(result, amount, price, &mpd_ctx);
-            if (mpd_cmp(result, taker->left, &mpd_ctx) > 0) {
-                mpd_set_i32(result, -m->stock_prec, &mpd_ctx);
-                mpd_pow(result, mpd_ten, result, &mpd_ctx);
-                mpd_sub(amount, amount, result, &mpd_ctx);
-            } else {
-                break;
+        if (bid_amount_money) {
+            mpd_div(amount, taker->left, price, &mpd_ctx);
+            mpd_rescale(amount, amount, -m->stock_prec, &mpd_ctx);
+            while (true) {
+                mpd_mul(result, amount, price, &mpd_ctx);
+                if (mpd_cmp(result, taker->left, &mpd_ctx) > 0) {
+                    mpd_set_i32(result, -m->stock_prec, &mpd_ctx);
+                    mpd_pow(result, mpd_ten, result, &mpd_ctx);
+                    mpd_sub(amount, amount, result, &mpd_ctx);
+                } else {
+                    break;
+                }
             }
-        }
 
-        if (mpd_cmp(amount, maker->left, &mpd_ctx) > 0) {
-            mpd_copy(amount, maker->left, &mpd_ctx);
-        }
-        if (mpd_cmp(amount, mpd_zero, &mpd_ctx) == 0) {
-            break;
+            if (mpd_cmp(amount, maker->left, &mpd_ctx) > 0) {
+                mpd_copy(amount, maker->left, &mpd_ctx);
+            }
+            if (mpd_cmp(amount, mpd_zero, &mpd_ctx) == 0) {
+                break;
+            }
+        } else {
+            if (mpd_cmp(taker->left, maker->left, &mpd_ctx) < 0) {
+                mpd_copy(amount, taker->left, &mpd_ctx);
```

Comment:
(More of a stylistic and lower priority comment than anything else.)

For a file of this size, could we not replace many of the if/else blocks with a ternary operator, to improve readability?

Reply:
sure but perhaps that kind of stylistic change should probably be in a dedicated PR